src/Lean/Meta/Tactic/AC/Sharing.lean

```diff
@@ -282,10 +284,68 @@ def canonicalizeEqWithSharing (ty lhs rhs : Expr) : SimpM Simp.Step := do
       proof? := some proof
     }
 
+
+theorem beq_congr {α : Type u} [inst : BEq α] {a₁ b₁ a₂ b₂ : α} (h₁ : a₁ = a₂) (h₂ : b₁ = b₂) :
+  (a₁ == b₁) = (a₂ == b₂) := by
```

Comment:
```suggestion
    (a₁ == b₁) = (a₂ == b₂) := by
```
formatting

Reply:
Also, I wonder if we couldn't just use a more generic `congrArg2`, like [Batteries](https://leanprover-community.github.io/mathlib4_docs/Batteries/Logic.html#congrArg%E2%82%82) has, and instantiate it with `f := @Eq \a` or `f := @BEq \a`

Reply:
sure! working on it together with the deduplication